### DISCUSSION TITLE
Advice for solving QP in parallel environments like Gym

### DISCUSSION THREAD

**Discussion by TRo202** [osqp/osqp#644] [General]
Is there anyone who has solved QP in parallel environments like Gym before?

What I want to do is solving a QP problem in each environment in parallel. I've been using OSQP only in a single environment in Gazebo, but couldn't find much information relating to using OSQP in multi environments in parallel. (#553 is the only related discussion I've found but still didn't get much information from here.)

So I would really love to get any tips or suggestions about how to solve QP in parallel environments. Thanks in advance!!

+edited)
I tagged the wrong discussion so I corrected the number.

**Comment by imciner2** [comment]
Which version of OSQP are you interested in using?

OSQP should be capable of being used in parallel environments with no changes, as long as you have a different solver object for each problem you wish to solve and also manually create the threads and call the OSQP library from the threads. 

There is a known issue with the upcoming GPU backend and multiple solvers at the same time, but the built-in algebra (which is the default, and also what is in the current 0.6 releases) should work fine.

**Reply by TRo202** [reply]
Thanks for your comment!

I'm currently using 0.6.3. 

May I ask how to create multiple threads and call OSQP from the threads? Is there any setting that I need to be aware of? I would greatly appreciate if you could kindly provide a brief code example for reference!

**Reply by imciner2** [reply]
The threading system you use will depend on the libraries you have available in Gym and what it supports, so I can't really provide good examples of that.

**Reply by TRo202** [reply]
Oh I see. Then could you please share any example for reference? I have no experience in running OSQP in multi threads...

Thank you in advance!
